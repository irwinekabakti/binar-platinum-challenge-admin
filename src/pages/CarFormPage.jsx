import React from "react";
import NavbarAdmin from "../components/Navbar/Navbar";
import CarForm from "../components/CarForm/CarForm";
import { Col, Container, Row } from "react-bootstrap";
import classes from "./CarFormPage.module.css";

const CarFormPage = ({ currentPage }) => {
  return (
    <Container fluid className={`p-0 m-0 ${classes.containerCarFormPage}`}>
      <NavbarAdmin currentPage="cars" />
      <Row className="m-0">
        <Col
          xs="auto"
          className={`d-none d-md-block h-100 ${classes.colCarFormPage}`}></Col>
        <Col className="px-3">
          <p className="mt-4">
            <strong>
              Cars {">"} List Car {">"}
            </strong>
            {currentPage === "edit" ? "Edit Car" : "Add New Car"}
          </p>
          <h5 className="mt-4 mb-3">
            <strong>
              {currentPage === "edit" ? "Edit Car" : "Add New Car"}
            </strong>
          </h5>
          <CarForm formFunction={currentPage === "edit" ? "edit" : "add"} />
        </Col>
      </Row>
    </Container>
  );
};

export default CarFormPage;
