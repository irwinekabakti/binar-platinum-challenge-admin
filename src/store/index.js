import { configureStore } from "@reduxjs/toolkit";

import authAdminSlice from "./action/admin-slice";
import dashboardSlice from "./action/dashboard-slice";

export const store = configureStore({
  reducer: {
    adminStore: authAdminSlice.reducer,
    dashboardStore: dashboardSlice.reducer,
  },
});
