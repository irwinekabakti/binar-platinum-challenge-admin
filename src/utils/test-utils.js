import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import authAdminSlice from "../store/action/admin-slice";
import dashboardSlice from "../store/action/dashboard-slice";
import { BrowserRouter } from "react-router-dom";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: {
        adminStore: authAdminSlice.reducer,
        dashboardStore: dashboardSlice.reducer,
      },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <BrowserRouter>
        <Provider store={store}>{children}</Provider>
      </BrowserRouter>
    );
  }
  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
