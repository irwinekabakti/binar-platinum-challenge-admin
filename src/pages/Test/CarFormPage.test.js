import { describe, test, expect } from "@jest/globals";
import Renderer from "react-test-renderer";
import { render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { renderWithProviders } from "../../utils/test-utils";
import { store } from "../../store";
import CarFormPage from '../CarFormPage'

describe('UI CarFormPage Test', () => {
  test('UI UI CarFormPage should match snapshot', () => {
    const snapshot = Renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <CarFormPage />
        </BrowserRouter>
      </Provider>
    ).toJSON()

    expect(snapshot).toMatchSnapshot()
  })
})
