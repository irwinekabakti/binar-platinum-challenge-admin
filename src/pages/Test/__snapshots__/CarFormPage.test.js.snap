// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI CarFormPage Test UI UI CarFormPage should match snapshot 1`] = `
<div
  className="p-0 m-0 containerCarFormPage container-fluid"
>
  <div
    className="bg-white navigation-panel position-absolute h-100  navPanelClass"
  >
    <div
      className="cars-menu w-100 flex-column"
    >
      <p
        className="w-100 ps-4 pe-3 titleCars"
      >
        CARS
      </p>
      <ul
        className="p-0 listCars"
      >
        <li
          className="ps-4 pe-3 listNameCars"
        >
          List Car
        </li>
      </ul>
    </div>
  </div>
  <div
    className="navigation-menu position-absolute flex-column  navMenuClass  nav"
    onKeyDown={[Function]}
  >
    <a
      className="navigationTab w-100 d-flex flex-column justify-content-center align-items-center text-center false nav-link"
      data-rr-ui-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "height": "64px",
        }
      }
      tabIndex={0}
    >
      <img
        alt="dashboard"
        src="fi_home.svg"
      />
      <p
        className="m-0 text-white navTitle"
      >
        Dashboard
      </p>
    </a>
    <a
      className="navigation-tab w-100 d-flex flex-column justify-content-center align-items-center text-center selected nav-link"
      data-rr-ui-event-key={null}
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "height": "64px",
        }
      }
      tabIndex={0}
    >
      <img
        alt="cars"
        src="fi_truck.svg"
      />
      <p
        className="m-0 text-white navTitle"
      >
        Cars
      </p>
    </a>
  </div>
  <nav
    className="bg-white p-0 mainNavbar navbar navbar-expand navbar-light"
  >
    <div
      className="h-100 p-0 d-flex justify-content-between align-items-center container-fluid"
    >
      <div
        className="h-100 d-flex colNav col-auto"
      >
        <button
          aria-label="Navigation Menu"
          className="menu-button d-flex justify-content-center align-items-center position-relative btnNavSideBar btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <div
            className="iconSideNav"
          />
        </button>
        <div
          className="h-100 w-100 d-flex align-items-center px-4 spaceNavTop"
        >
          <div
            className="iconTopNav"
          />
        </div>
      </div>
      <div
        className="h-100 d-flex justify-content-between col"
      >
        <div
          className="h-100 w-100 d-flex justify-content-center align-items-center containerBurgerButton"
          data-testid="container-burger-button-custom"
        >
          <button
            className="bg-white p-1 d-flex flex-column border-0 btnBurgerButton btn btn-primary"
            data-testid="burger-button-custom"
            disabled={false}
            type="button"
          >
            <span
              className="bg-dark burgerButton"
            />
            <span
              className="bg-dark burgerButton"
            />
            <span
              className="bg-dark burgerButton"
            />
          </button>
        </div>
        <div
          className="d-flex h-100"
        >
          <form
            className="d-none d-md-flex align-items-center pe-3"
          >
            <fieldset
              className="position-relative d-flex"
            >
              <input
                className="formNavbar formNavbar form-control"
                placeholder="Search"
                type="text"
              />
              <img
                alt="search"
                aria-hidden="true"
                className="iconSearch"
                src="fi_search.svg"
              />
              <button
                className="fw-bold btnNavbarSearch btn btn-primary"
                disabled={false}
                type="button"
              >
                Search
              </button>
            </fieldset>
          </form>
          <div
            className="h-100 w-100 pe-3 d-flex justify-content-center align-items-center containerProfile"
          >
            <div
              className="dropdown nav-item"
            >
              <button
                aria-expanded={false}
                className="bg-white border-0 d-flex align-items-center admin-menu dropdown-toggle btn btn-primary"
                disabled={false}
                id="dropdown-1"
                onClick={[Function]}
                type="button"
              >
                <div
                  className="rounded-circle d-flex justify-content-center align-items-center ImageProfileNavbar"
                >
                  R
                </div>
                <div
                  className="px-2 text-dark profileNavbar"
                >
                  Marcus Raspot
                </div>
                <img
                  alt="icon-navbar"
                  aria-hidden="true"
                  src="chevron-down.svg"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="m-0 row"
  >
    <div
      className="d-none d-md-block h-100 colCarFormPage col-auto"
    />
    <div
      className="px-3 col"
    >
      <p
        className="mt-4"
      >
        <strong>
          Cars 
          &gt; 
           List Car 
          &gt; 
        </strong>
        Add New Car
      </p>
      <h5
        className="mt-4 mb-3"
      >
        <strong>
          Add New Car
        </strong>
      </h5>
      <form
        className=""
        onSubmit={[Function]}
      >
        <div
          className="w-100 bg-white p-3"
        >
          <fieldset
            className="fieldFormSet w-100"
          >
            <div
              className="mb-3 row"
              data-testid="wrapper-labelName"
            >
              <label
                className="mb-0 d-flex align-items-center form-label col-form-label col-sm-4"
                data-testid="label-Name"
                htmlFor="name"
              >
                Nama/Tipe Mobil
                <span
                  className="text-danger"
                  data-testid="label-SpanName"
                >
                  *
                </span>
              </label>
              <div
                className="col-sm-8"
              >
                <input
                  className="formBox form-control"
                  id="name"
                  onChange={[Function]}
                  placeholder="Input Nama/Tipe Mobil"
                  required={true}
                  type="text"
                />
              </div>
            </div>
            <div
              className="mb-3 row"
              data-testid="wrapper-labelPrice"
            >
              <label
                className="mb-0 d-flex align-items-center form-label col-form-label col-sm-4"
                data-testid="label-Price"
                htmlFor="price"
              >
                Harga
                <span
                  className="text-danger"
                  data-testid="label-SpanPrice"
                >
                  *
                </span>
              </label>
              <div
                className="col-sm-8"
              >
                <input
                  className="formBox form-control"
                  id="price"
                  onChange={[Function]}
                  placeholder="Input Harga Sewa Mobil"
                  required={true}
                  type="number"
                />
              </div>
            </div>
            <div
              className="mb-3 row"
              data-testid="wrapper-Photo"
            >
              <label
                className="mb-0 d-flex align-items-center form-label col-form-label col-sm-4"
                data-testid="label-Photo"
                htmlFor="image"
              >
                Foto
                <span
                  className="text-danger"
                  data-testid="label-SpanPhoto"
                >
                  *
                </span>
              </label>
              <div
                className="position-relative col-sm-8"
              >
                <img
                  alt=""
                  className="position-absolute"
                  src="fi_upload.svg"
                  style={
                    Object {
                      "right": "21px",
                      "top": "9px",
                    }
                  }
                />
                <input
                  accept="image/png, image/gif, image/jpeg"
                  className="formBox form-control"
                  id="image"
                  onChange={[Function]}
                  required={true}
                  type="file"
                />
                <p
                  className="mb-0"
                  style={
                    Object {
                      "color": "#8A8A8A",
                      "fontSize": "10px",
                      "lineHeight": "14px",
                      "lineWeight": 300,
                    }
                  }
                >
                  File size max. 2MB
                </p>
              </div>
            </div>
            <div
              className="mb-3 row"
              data-testid="wrapper-Category"
            >
              <label
                className="mb-0 d-flex align-items-center form-label col-form-label col-sm-4"
                data-testid="label-Category"
                htmlFor="category"
              >
                Kategori
                <span
                  className="text-danger"
                  data-testid="label-SpanCategory"
                >
                  *
                </span>
              </label>
              <div
                className="col-sm-8"
              >
                <select
                  className="formBox form-select"
                  defaultValue={null}
                  id="category"
                  onChange={[Function]}
                >
                  <option
                    hidden={true}
                  >
                    Pilih Kategori Mobil
                  </option>
                  <option
                    value="small"
                  >
                    2 - 4 orang
                  </option>
                  <option
                    value="medium"
                  >
                    4 - 6 orang
                  </option>
                  <option
                    value="large"
                  >
                    6 - 8 orang
                  </option>
                </select>
              </div>
            </div>
            <div
              className="formInfo"
            >
              <div
                className="mb-3 row"
              >
                <div
                  className="mb-0 col-sm-4"
                >
                  Created at
                </div>
                <div
                  className="col-sm-8"
                >
                  -
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="mb-0 col-sm-4"
                >
                  Updated at
                </div>
                <div
                  className="col-sm-8"
                >
                  -
                </div>
              </div>
            </div>
          </fieldset>
        </div>
        <div
          className="d-flex"
          style={
            Object {
              "marginTop": "40px",
            }
          }
        >
          <button
            className="d-flex align-items-center me-3 formButtonCancel btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="d-flex align-items-center text-white formButtonSave btn btn-primary"
            disabled={false}
            type="submit"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
